src/test/suite/helper.ts

```diff
@@ -47,43 +80,813 @@ export function mockTextDocument(uri: Uri, languageId: string, source: string[])
+    await waitForCondition(
+        async () => assertHasTextOutputInVSCode(cell, text, index, isExactMatch),
+        timeout,
+        () =>
+            `Output does not contain provided text '${text}' for Cell ${cell.index + 1}, it is ${getCellOutputs(cell)}`
+    );
+}
+export async function saveActiveNotebook() {
+    await vscode.commands.executeCommand('workbench.action.files.saveAll');
+}
+
+export function noop() {
+    // Do nothing
+}
+
+export function traceInfo(...args: any[]) {
+    console.log(args);
+}
+
+export const NotebookCellScheme = 'vscode-notebook-cell';
+export const InteractiveInputScheme = 'vscode-interactive-input';
+export const InteractiveScheme = 'vscode-interactive';
+
+export const PYTHON = [
+    { scheme: 'file', language: PYTHON_LANGUAGE },
+    { scheme: 'untitled', language: PYTHON_LANGUAGE },
+    { scheme: 'vscode-notebook', language: PYTHON_LANGUAGE },
+    { scheme: NotebookCellScheme, language: PYTHON_LANGUAGE },
+    { scheme: InteractiveInputScheme, language: PYTHON_LANGUAGE }
+];
+
+let languageClient: vslc.LanguageClient | undefined = undefined;
+let languageClientDisposable: vscode.Disposable | undefined = undefined;
+let cancellationStrategy: FileBasedCancellationStrategy | undefined = undefined;
+
+function ensure(target: any, key: string) {
+    if (target[key] === undefined) {
+        target[key] = {};
+    }
+    return target[key];
+}
+
+class NerfedExecuteCommandFeature implements DynamicFeature<ExecuteCommandRegistrationOptions> {
+    private _commands: Map<string, vscode.Disposable[]> = new Map<string, vscode.Disposable[]>();
+
+    constructor() {}
+
+    public get registrationType(): RegistrationType<ExecuteCommandRegistrationOptions> {
+        return ExecuteCommandRequest.type;
+    }
+
+    public fillClientCapabilities(capabilities: ClientCapabilities): void {
+        ensure(ensure(capabilities, 'workspace'), 'executeCommand').dynamicRegistration = true;
+    }
+
+    public initialize(capabilities: ServerCapabilities): void {
+        if (!capabilities.executeCommandProvider) {
+            return;
+        }
+        this.register({
+            id: uuid(),
+            registerOptions: Object.assign({}, capabilities.executeCommandProvider)
+        });
+    }
+
+    public register(_data: RegistrationData<ExecuteCommandRegistrationOptions>): void {
+        // Do nothing. Otherwise we end up with double registration
+        traceInfo('Registering dummy command feature');
+    }
+
+    public unregister(id: string): void {
+        let disposables = this._commands.get(id);
+        if (disposables) {
+            disposables.forEach((disposable) => disposable.dispose());
+        }
+    }
+
+    public dispose(): void {
+        this._commands.forEach((value) => {
+            value.forEach((disposable) => disposable.dispose());
+        });
+        this._commands.clear();
+    }
+}
+
+async function startLanguageServer(languageServerFolder: string, pythonPath: string) {

```

Comment:
This is the code that mimics what the python extension does. It uses the language server bundle from pylance to start a language server. It also wires up the middleware piece that this NPM module exports.